src/updatechecker.cpp

```diff
@@ -219,52 +220,80 @@ UpdateChecker::UpdateChecker(): Thread("WinSparkle updates check")
 
 void UpdateChecker::Run()
 {
-    // no initialization to do, so signal readiness immediately
-    SignalReady();
-
-    try
-    {
-        const std::string url = Settings::GetAppcastURL();
-        if ( url.empty() )
-            throw std::runtime_error("Appcast URL not specified.");
-        CheckForInsecureURL(url, "appcast feed");
-
-        StringDownloadSink appcast_xml;
-        DownloadFile(url, &appcast_xml, this, GetAppcastDownloadFlags());
-
-        Appcast appcast = Appcast::Load(appcast_xml.data);
-        if (!appcast.ReleaseNotesURL.empty())
-            CheckForInsecureURL(appcast.ReleaseNotesURL, "release notes");
-        if (!appcast.DownloadURL.empty())
-            CheckForInsecureURL(appcast.DownloadURL, "update file");
-
-        Settings::WriteConfigValue("LastCheckTime", time(NULL));
-
-        const std::string currentVersion =
-                WideToAnsi(Settings::GetAppBuildVersion());
-
-        // Check if our version is out of date.
-        if ( !appcast.IsValid() || CompareVersions(currentVersion, appcast.Version) >= 0 )
-        {
-            // The same or newer version is already installed.
-            UI::NotifyNoUpdates(ShouldAutomaticallyInstall());
-            return;
-        }
-
-        // Check if the user opted to ignore this particular version.
-        if ( ShouldSkipUpdate(appcast) )
-        {
-            UI::NotifyNoUpdates(ShouldAutomaticallyInstall());
-            return;
-        }
+	while (1)
+	{
+		// no initialization to do, so signal readiness immediately
+ 		SignalReady();
+
+		bool checkUpdates;
+		if (Settings::ReadConfigValue("CheckForUpdates", checkUpdates))
+		{
+			if (checkUpdates)
+			{
+				static const time_t ONE_DAY = 60 * 60 * 24;
+
+				time_t lastCheck = 0;
+				Settings::ReadConfigValue("LastCheckTime", lastCheck);
+				const time_t currentTime = time(NULL);
+
+				// Only check for updates in reasonable intervals:
+				const int interval = win_sparkle_get_update_check_interval();
+				if (currentTime - lastCheck >= interval)
+				{
+					UpdateCheckWorker();
+				}
+			}
+		}
+		// Check every 5 minutes
+		Sleep(300000);
```

Comment:
Why is the high frequency (every 5 minutes) periodic checking needed? Why can't we just sleep once for the exact time defined by the returned value of win_sparkle_get_update_check_interval()?